python/test/t_CovarianceModel_std.py

```diff
@@ -3,73 +3,90 @@
+def test_model(myModel, test_partial_grad=True, x1=None, x2=None):
 
-    print('myModel = ',  myModel)
 
     inputDimension = myModel.getInputDimension()
     dimension = myModel.getOutputDimension()
-    active = myModel.getActiveParameter()
-    print('active=', active)
-    print('parameter=', myModel.getParameter())
-    print('parameterDescription=', myModel.getParameterDescription())
 
     if x1 is None and x2 is None:
         x1 = ot.Point(inputDimension)
         x2 = ot.Point(inputDimension)
         for j in range(inputDimension):
             x1[j] = -1.0 - j
             x2[j] = 3.0 + 2.0 * j
-
-    eps = 1e-5
-    print('myModel(', x1, ', ', x2, ')=',  repr(myModel(x1, x2)))
-
-    grad = myModel.partialGradient(x1, x2)
-    print('dCov =', repr(grad))
-
-    if (dimension == 1):
-        gradfd = ot.Point(inputDimension)
-        for j in range(inputDimension):
-            x1_d = ot.Point(x1)
-            x1_d[j] = x1_d[j] + eps
-            gradfd[j] = (myModel.computeAsScalar(x1_d, x2) - myModel.computeAsScalar(x1, x2)) / eps
     else:
-        gradfd = ot.Matrix(inputDimension, dimension * dimension)
-        covarianceX1X2 = myModel(x1, x2)
-        centralValue = ot.Point(covarianceX1X2.getImplementation())
-        # Loop over the shifted points
-        for i in range(inputDimension):
-            currentPoint = ot.Point(x1)
-            currentPoint[i] += eps
-            localCovariance = myModel(currentPoint, x2)
-            currentValue = ot.Point(
-                localCovariance.getImplementation())
-            for j in range(currentValue.getSize()):
-                gradfd[i, j] = (currentValue[j] - centralValue[j]) / eps
-    print('dCov (FD)=', repr(gradfd))
-
-    if test_grad:
-        pGrad = myModel.parameterGradient(x1, x2)
-        precision = ot.PlatformInfo.GetNumericalPrecision()
-        ot.PlatformInfo.SetNumericalPrecision(4)
-        print('dCov/dP=', pGrad)
-        ot.PlatformInfo.SetNumericalPrecision(precision)
-
-
-def test_scalar_model(myModel):
-
-    inputDimension = 1
-    dimension = 1
-    active = myModel.getActiveParameter()
-    
-    x1 = 2.0
-    x2 = -3.0
+        x1 = ot.Point(x1)
+        x2 = ot.Point(x2)
+
+    if myModel.isStationary():
+        ott.assert_almost_equal(myModel(x1 - x2), myModel(x1, x2), 1e-14, 1e-14)
+        ott.assert_almost_equal(myModel(x2 - x1), myModel(x1, x2), 1e-14, 1e-14)
+
+    eps = 1e-3
+
+    mesh = ot.IntervalMesher([9]*inputDimension).build(
+        ot.Interval([-10]*inputDimension, [10]*inputDimension))
+
+    C = myModel.discretize(mesh)
+    if dimension == 1:
+        # Check that discretize & computeAsScalar provide the 
+        # same values
+        vertices = mesh.getVertices()
+        for j in range(len(vertices)):
+            for i in range(j, len(vertices)):
+                ott.assert_almost_equal(C[i,j], myModel.computeAsScalar(vertices[i], vertices[j]), 1e-14, 1e-14)
```

Comment:
```suggestion
                ott.assert_almost_equal(C[i,j], myModel.computeAsScalar(vertices[i], vertices[j]), 0.0, 0.0)
```
Again exactly the same computation.